doc/manual/rl-next/stabilize-fetchtree.md

```diff
@@ -0,0 +1,9 @@
+---
+synopsis: Stabilize fetchTree
+prs: 10068
+---
+
+The core of `fetchTree` is now stable.
+This includes
+- the `fetchTree` function itself
+- all the existing fetchers, except `git` (still unstable because of some reproducibility concerns)
```

Comment:
`git` is the only one we've actually improved during this time.

`gitlab` has known problems (wrong field names, wrong url syntax)

`mercurial` and all the other fetchers we haven't talked about are under-used and should be assumed to have repro problems

`github` is still `git archive`-based, not tree-based.
 - users who switch between `git` and `github` fetchers will be surprised to see their hash changed
 - `git` can't be _reliably_ be used as a fallback implementation for `github` fetching because of `export-subst` (`export-ignore` *might* work)

`tarball` is probably the only fetcher without severe issues and also been reviewed properly. All the other ones should remain feature gated.

Reply:
Discussed during the Nix maintainers meeting on 2024-02-26.
Approved, but needs to be more restrictive, and blocked on https://github.com/NixOS/nix/issues/10115

<details>

- Stabilizing everything but git was too eager, we need to just stabilize the tarball one
- Should it return a path value as the output path?
    - Lazy trees require that
    - We can also just keep `fetchTree` not lazy
- Any reason not to return a path?
    - Will cause errors in some usages
- Any complexity in making it return a path?
    - Hopefully a one-line change
    - Already the case on lazy-trees
    - @edolstra can backport it from lazy-trees
        - https://github.com/NixOS/nix/issues/10115